src/Database.hs

```diff
@@ -0,0 +1,18 @@
+{-# LANGUAGE OverloadedStrings, ExtendedDefaultRules #-}
+module Database where
+import Database.MongoDB    (Action, Document, Document, Field, Value, Val, access,
+                            close, connect, delete, exclude, find, count,
+                            host, insertMany, master, project, rest,
+                            select, sort, aggregate, (=:), (!?), typed, value,
+                            ObjectId, Label, valueAt, modify, look)
+
+import Data.Function (on)
+import Data.List (sortBy, groupBy)
+import Data.Ord (comparing)
+import Data.Text (pack)
+import Control.Monad.Trans (liftIO)
+
+runMongo dbName functionToRun = do
```

Comment:
All these changes look good to me. I think what remains is to call this on post to the API with a `functionToRun` that performs the insert. This is the tricky part, as it involves the interaction of some monads. From what I can see from quickly looking at the docs, the `Database.MongoDB.Action` monad requires a `MonadIO`, which the `Snap` monad is an instence of, so maybe there's a way for them to interact there. At the moment, I only have a vague notion of how this interaction would work. What would you think about pairing up on it some evening during the week, if you're free?